src/cdk/observers/observe-content.ts

```diff
@@ -35,6 +32,70 @@ export class MutationObserverFactory {
   }
 }
 
+
+/** A factory that creates ContentObservers. */
+@Injectable({providedIn: 'root'})
+export class ContentObserver {
+  private _observedElements = new Map<Element, {
+    observer: MutationObserver | null,
+    stream: Subject<MutationRecord[]>,
+    count: number
+  }>();
+
+  constructor(private _mutationObserverFactory: MutationObserverFactory) {}
+
+  observe(element: Element, debounce?: number): Observable<MutationRecord[]> {
```

Comment:
What's the reasoning behind making `debouce` part of the API rather than letting subscribers debounce (or throttle or filter, etc) however they want?

Reply:
I put it here because it was part of the directive's logic, but I agree it doesn't really belong here, will remove.